examples/sensor-websocket/src/components/SensorWebSocket.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { Component } from "react";
+import { getChannelToken, getWSUrl, socketParams } from "../apiHandler";
+import {
+  connection_trigger_payload,
+  disconnection_trigger_payload,
+  value_trigger_payload
+} from "../triggerPayload";
+import { Col, Container, ListGroup, Row } from "react-bootstrap";
+import moment from "moment";
+
+const { Socket } = require("phoenix");
+
+const connection = {
+  device_connected: {
+    name: "Device Connected",
+    color: "#008000"
+  },
+  device_disconnected: {
+    name: "Device Disconnected",
+    color: "#ff0000"
+  }
+};
+
+const listItem = (item, index) => {
+  return (
+    <ListGroup.Item
+      key={index}
+      className="temperature-list px-3 border-0 pb-0 py-2 font-weight-normal bg-transparent"
+    >
+      <span className="font-weight-light">
+        <strong className="font-weight-bold">{item.label}: </strong>
+        {item.value}
+      </span>
+    </ListGroup.Item>
+  );
+};
+
+const setSensorDataCard = (index, sensor) => {
+  const sensor_item = [
+    {
+      label: "Value",
+      value: sensor.value
+    },
+    {
+      label: "Last Update",
+      value: sensor.timestamp
+    }
+  ];
+  return (
+    <Row key={index} className="m-3 bg-light px-2 pt-3 pb-2">
+      <Col className="text-dark px-3 py-2 bg-transparent border-0 font-weight-bold w-100 text-left btn text-decoration-none">
+        <span className="bg-sensor-theme font-weight-light py-2 px-3 w-100 text-white d-inline-block">
+          <strong className="font-weight-bold">Sensor Name : &nbsp;</strong>
+          {sensor.name}
+        </span>
+        <Col className="p-0">
+          <ListGroup className="py-1 my-2">
+            {sensor_item.map(listItem)}
+          </ListGroup>
+        </Col>
+      </Col>
+    </Row>
+  );
+};
+
+const connectionStatus = (sensors, device) => {
+  return (
+    <Container className="mt-2 p-0 bg-transparent device-status-div">
+      <Row className="m-3">
+        <Col className="col-12 text-dark px-3 py-2 bg-transparent border-0 w-100 text-left btn text-decoration-none">
+          <h6 className="m-0 font-weight-bold position-relative">
+            Device Status :{" "}
+            <span
+              className="status-tag rounded-circle d-inline-block mr-2"
+              style={{
+                backgroundColor: connection[device.connection_type].color
+              }}
+            />
+            {connection[device.connection_type].name}
+          </h6>
+        </Col>
+      </Row>
+      {Object.keys(sensors).map((key, index) => {
+        const sensor = sensors[key];
+        return setSensorDataCard(index, sensor);
+      })}
+    </Container>
+  );
+};
+
+const errorSocket = () => {
+  return <p>Socket is giving an Error</p>;
+};
+
+const closeSocket = () => {
+  return <p>Socket is Closed</p>;
```

Comment:
The user doesn't, and shoudn't know that the data is retrieved using a socket. All the user cares is getting his device info. Let's show meaningful error messages when we can't accomplish  what the user asked